lsp/nls/src/linearization.rs

```diff
@@ -347,11 +517,28 @@ impl Linearizer<BuildingResource, (UnifTable, HashMap<usize, Ident>)> for Analys
     /// Additionally, resolves concrete types for all items.
     fn linearize(
         self,
-        lin: Linearization<Building<BuildingResource>>,
+        mut lin: Linearization<Building<BuildingResource>>,
         (table, reported_names): (UnifTable, HashMap<usize, Ident>),
     ) -> Linearization<Completed> {
-        let mut lin_ = lin.state.resource.linearization;
         eprintln!("linearizing");
+
+        let defers: Vec<(usize, usize, Ident)> = lin
```

Comment:
Does this mean we make a full pass on the linearization just to find potential deferred usage? Would there be a way to remember indices of deferred item along the way to avoid this? Not saying this should be done in this PR, but just to know.

Reply:
Yes right now its filtering the linearization for deferred usages. I could track them in the building type if filtering them is too much work.

Reply:
Its technically in the nls crate (by implementation). It becomes very mixed up with nickel and nls as I would need to pass down the trace identifier (currently Req. Id) but that looks to create more trouble than it solves maybe

Reply:
Got it, let's forget it for now then.